### DISCUSSION TITLE
Flet App - "Open with" functionality

### DISCUSSION THREAD

**Discussion by amcraig** [flet-dev/flet#2635] [Q&A]
### Question

Hi all, I was curious if anyone has built this in, or there is an official method, but is there a way to build a Flet app such that the user can open a file with an approved file extension immediately on start as opposed to having to use a file picker dialog.

The exact functionality that I'm looking for is (at least for me on MacOS) is to find a file of interest in the file explorer, right click the file, and then select the "Open with..." option. 

This would be tremendously helpful to my team.

Thanks in advance!

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by gjelsas** [comment]
I would concur @amcraig. This would be a nice feature. 

I would like to create a .ics-file after a certain operation, and it would be nice to just import it into Thunderbird or whatever.

**Comment by ingdesarr1** [comment]
I work in Windows, and I use the library subprocess to open files that I saved. I don't know if this library works on mac, but maybe can work.

**Comment by gjelsas** [comment]
I wish there would be a way to use the flet framework for this. Then it would work on mobile, the web and Linux/Mac/Windows alike...
